src/vrf.c

```diff
@@ -0,0 +1,289 @@
+//
+// Created by Tuposoft Collective on 23.01.2023.
+//
+
+#include <arpa/nameser.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <resolv.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include "vrf.h"
+
+#ifndef FREE
+#define FREE(p) free(p); p = NULL;
+#endif
+
+#define CHECK_OK(f, err)    \
+if ((err = f) != VRF_OK) {  \
+    return err;             \
+}
+
+struct
+VRF {
+    char *email;
+    char *local_part;
+    char *domain;
+    char *mx_record;
+    char *mx_domain;
+    bool result;
+    bool catch_all;
+};
+
+static VRF_err_t
+send_command(int sock, char *format, ...)
+{
+    va_list args;
+    va_start(args, format);
+    char *command;
+    if (vasprintf(&command, format, args) < 0) {
```

Comment:
Should we really use `vasprintf`? Judging by the documentation [here](https://linux.die.net/man/3/vasprintf), it says that it allocates memory, but you never free it so I think this causes a memory leak.

Reply:
Oh never mind, you free it, sorry :smiley: